python/lsst/afw/geom/linearTransform.cc

```diff
@@ -59,37 +62,50 @@ PYBIND11_PLUGIN(_linearTransform) {
     clsLinearTransform.def(py::init<typename LinearTransform::Matrix const &>());
 
     /* Operators */
-    clsLinearTransform.def("__getitem__", [](LinearTransform const & self, int i) {
-            return self[lsst::utils::cppIndex(4, i)];
-        }, py::is_operator());
-    clsLinearTransform.def("__getitem__", [](LinearTransform const & self, std::pair<int, int> i) {
-            auto row = lsst::utils::cppIndex(2, i.first);
-            auto col = lsst::utils::cppIndex(2, i.second);
-            return self.getMatrix()(row, col);
-        }, py::is_operator());
-
+    clsLinearTransform.def("__call__",
+                           (Point2D(LinearTransform::*)(Point2D const &) const) & LinearTransform::operator(),
+                           py::is_operator());
+    clsLinearTransform.def(
+        "__call__", (Extent2D(LinearTransform::*)(Extent2D const &) const) & LinearTransform::operator(),
+        py::is_operator());
+    clsLinearTransform.def(
+        "__getitem__", [](LinearTransform const &self, int i) { return self[lsst::utils::cppIndex(4, i)]; },
+        py::is_operator());
+    clsLinearTransform.def("__getitem__",
+                           [](LinearTransform const &self, std::pair<int, int> i) {
+                               auto row = lsst::utils::cppIndex(2, i.first);
+                               auto col = lsst::utils::cppIndex(2, i.second);
```

Comment:
Originally I had it templated on an STL container type and used `size()` but then it would not work with `afw:geom::Point`, `ndarray` arrays, `Eigen` matrices... so that was a bust. Given the lack of a standard way to get size or dimensions I think taking the size as an int-like argument is best. I take your point about magic numbers, but I feel they are fairly harmless in this context.

Note that this is pre-existing code. I just ran it through clang-format because some of it was too long.